test/cli/watch.js

```diff
@@ -0,0 +1,248 @@
+"use strict";
+
+const fs = require( "fs-extra" );
+const path = require( "path" );
+const exec = require( "execa" ).shell;
+const co = require( "co" );
+const fixturify = require( "fixturify" );
+
+const expectedWatchOutput = require( "./fixtures/expected/watch-tap-outputs" );
+
+// Executes the provided command from within the fixtures directory
+function execute( command ) {
+	const cwd = process.cwd();
+	process.chdir( path.join( __dirname, "fixtures" ) );
+
+	const execution = exec( command, { stdio: [ null, null, null, "ipc" ] } );
+
+	process.chdir( cwd );
+
+	return execution;
+}
+
+const fixturePath = path.join( __dirname, "fixtures", "watching" );
+
+// Kills the provided executing process, handling differences in platforms
+function kill( execution, signal ) {
+	const sig = signal || "SIGINT";
+
+	// Linux increments the pid by 1 due to creating a new shell instance
+	if ( process.platform === "linux" ) {
```

Comment:
We could add more logic and special case per-platform, but I'd rather keep this simple to start and evolve it as needed over time.

The kill-with-child thing isn't really applicable, because we don't want to kill the parent process (which is the test suite running).